src/main/java/seedu/address/logic/commands/AddTaskCommand.java

```diff
@@ -41,29 +40,23 @@ public class AddTaskCommand extends Command {
 
     public static final String MESSAGE_SUCCESS = "New task added: %1$s";
     public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in the address book";
-
     private final Task toAdd;
-    private final Index personIndex;
+    private final Email personEmailAddress;
 
     /**
      * Creates an AddTaskCommand to add the specified {@code Task}
      */
-    public AddTaskCommand(Task task, Index personIndex) {
+    public AddTaskCommand(Task task, Email personEmailAddress) {
         requireNonNull(task);
         toAdd = task;
-        this.personIndex = personIndex;
+        this.personEmailAddress = personEmailAddress;
     }
```

Comment:
looks good overall! thanks for changing the test to suit change from personIndex to personEmailAddress